internal/exec/exec.go

```diff
@@ -201,73 +266,19 @@ func execFieldSelection(ctx context.Context, r *Request, s *resolvable.Schema, f
 			return errors.Errorf("%s", err) // don't execute any more resolvers if context got cancelled
 		}
 
-		res := f.resolver
-		if f.field.UseMethodResolver() {
-			var args interface{}
-
-			if f.field.ArgsPacker != nil {
-				args = f.field.PackedArgs.Interface()
-			}
-
-			resolver := directives.ResolverFunc(func(ctx context.Context, args interface{}) (output interface{}, err error) {
-				var in []reflect.Value
-				var callOut []reflect.Value
-
-				if f.field.HasContext {
-					in = append(in, reflect.ValueOf(ctx))
-				}
-
-				if f.field.ArgsPacker != nil {
-					in = append(in, reflect.ValueOf(args))
-				}
-
-				callOut = res.Method(f.field.MethodIndex).Call(in)
-				result = callOut[0]
-
-				if f.field.HasError && !callOut[1].IsNil() {
-					resolverErr := callOut[1].Interface().(error)
-					return result.Interface(), resolverErr
-				}
-
-				return result.Interface(), nil
-			})
-
-			res, resolverErr := r.executeWithDirectives(traceCtx, f, args, resolver)
-
-			if resolverErr != nil {
-				err := errors.Errorf("%s", resolverErr)
-				err.Path = path.toSlice()
-				err.ResolverError = resolverErr
-				if ex, ok := resolverErr.(extensionser); ok {
-					err.Extensions = ex.Extensions()
-				}
-				return err
-			}
-
-			result = reflect.ValueOf(res)
-		} else {
-			// TODO extract out unwrapping ptr logic to a common place
-			if res.Kind() == reflect.Ptr {
-				res = res.Elem()
-			}
-
-			resolver := func(ctx context.Context, _ interface{}) (output interface{}, err error) {
-				return res.FieldByIndex(f.field.FieldIndex).Interface(), nil
+		res, resolverErr := f.resolve(ctx, r)
```

Comment:
This code is much cleaner to read and reason about!

Reply:
A slight performance degradation might be observed when executing a query with huge number of fields because [this function will not be inlined](https://stackoverflow.com/questions/45836981/is-it-possible-to-inline-function-containing-loop-in-golang). However, the readability significantly improved, IMHO.